image2tensor/src/image2tensor.rs

```diff
@@ -1,82 +1,97 @@
-use image::DynamicImage;
-use image::io::Reader;
+use image::{io::Reader, Bgr, DynamicImage, ImageBuffer, Rgb};
 
 #[derive(Debug, Copy, Clone)]
 pub enum TensorType {
     F16,
     F32,
     U8,
-    I32
+    I32,
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum ColorOrder {
+    RGB,
+    BGR,
 }
 
 // Take the image located at 'path', open it, resize it to height x width, and then converts
 // the pixel precision to the type requested.
-pub fn convert_image (
-    path: String,
+pub fn convert_image(
+    path: &String,
     width: u32,
     height: u32,
     precision: TensorType,
-    out_buff: &mut [u8]
-    ) -> i32 {
+    order: ColorOrder,
+) -> Vec<u8> {
     let raw_file = Reader::open(path);
     let raw_file = match raw_file {
         Ok(file) => file,
         Err(e) => {
             println!("Failed to open the file: {:?}", e);
-            return -1;
-        },
+            return vec![];
+        }
     };
 
-    let decoded = raw_file.decode(); //.unwrap();
+    let decoded = raw_file.decode();
     let decoded = match decoded {
         Ok(pixels) => pixels,
         Err(e) => {
             println!("Failed to decode the image: {:?}", e);
-            return -1;
-        },
+            return vec![];
+        }
     };
 
     let dyn_img: DynamicImage = decoded.resize_exact(width, height, image::imageops::Triangle);
-    // Switch from RGB to BGR
-    let bgr_img = dyn_img.to_bgr8();
+    let rgb_bind: ImageBuffer<Rgb<u8>, Vec<u8>>;
+    let bgr_bind: ImageBuffer<Bgr<u8>, Vec<u8>>;
 
     // Get an array of the pixel values
-    let raw_u8_arr: &[u8] = &bgr_img.as_raw()[..];
-    let bytes_per_pixel: usize = get_bytes_per_pixel(precision);
-    let total_bytes: i32 = (raw_u8_arr.len() * bytes_per_pixel) as i32;
-
-    // Create an array to hold the f32 value of those pixels
-    for i in 0..raw_u8_arr.len()  {
-
-        // Read the number as a f32 and break it into u8 bytes
-        let u8_f32: f32 = raw_u8_arr[i] as f32;
-        let u8_bytes = u8_f32.to_ne_bytes();
-
-        for j in 0..bytes_per_pixel {
-            out_buff[(i * bytes_per_pixel) + j] = u8_bytes[j];
+    let raw_u8_arr = match order {
+        ColorOrder::RGB => {
+            rgb_bind = dyn_img.to_rgb8();
+            rgb_bind.as_raw()
         }
+        ColorOrder::BGR => {
+            bgr_bind = dyn_img.to_bgr8();
```

Comment:
I think this function is going away in later versions of `image`: https://github.com/image-rs/image/releases/tag/v0.24.0.

Reply:
It is. But I'd like to address that in a future release if possible since there's really no reason to upgrade to the newer version just yet. And I haven't had time to dive into the new API.